C-Unit1/src/E01_Functions.cpp

```diff
@@ -0,0 +1,74 @@
+//
+//  E01_Functions.cpp
+//  Unit1_exercise1_understanding_functions
+//
+
+/*
+ 
+ Note:
+ Implement the functions in this exercise file to make the tests
+ pass in E01_FunctionsSpec.cpp file.
+ 
+ */
+
+
+//
+// Note:
+//  Think of all the edge cases.
+//
+// returns the mid-point of
+// given 2 points on X-axis
+//
+using namespace std;
+int midPoint(int x1, int x2) {
+	//cout << x1;
+	
+    return x1/2+x2/2+((x1%2!=0 &&x2%2!=0)?1:0);
+}
+
+//
+// sum of most significant digits of non-negative integers.
+//
+int sumOfMSDs(int num1, int num2) {
+	int a, b;
+	a = b = 0;
+	while (num1) {
+		a = num1;
+		num1 /= 10;
+	}
+	while (num2) {
+		b = num2;
+		num2 /= 10;
+	}
+    return a+b;
```

Comment:
i fell that having a common loop can reduce the number of computations .
but i have a little confusion that however the same number of computations are required for getting most significant digit.
Anna i feel like can you give us solutions to the questions after  we complete the  submissions with requested changes. so that we can learn different types of answering to the same questions.